src/ui.rs

```diff
@@ -3089,28 +3109,17 @@ impl Component for SectionLineView<'_> {
                 line,
             } => {
                 let toggle_box_rect = viewport.draw_component(x, y, toggle_box);
-                let x = x + toggle_box_rect.width.unwrap_isize() + 1;
+                let x = toggle_box_rect.end_x() + 1;
 
-                let (change_type_text, style) = match change_type {
+                let (change_type_text, changed_line_style) = match change_type {
                     ChangeType::Added => ("+ ", Style::default().fg(Color::Green)),
                     ChangeType::Removed => ("- ", Style::default().fg(Color::Red)),
                 };
-                viewport.draw_span(x, y, &Span::styled(change_type_text, style));
-                let x = x + change_type_text.width().unwrap_isize();
-                let (line, line_end) = match line.strip_suffix('\n') {
-                    Some(line) => (
-                        Span::styled(line, style),
-                        Some(Span::styled(
-                            NEWLINE_ICON,
-                            Style::default().fg(Color::DarkGray),
-                        )),
-                    ),
-                    None => (Span::styled(*line, style), None),
-                };
-                let line_rect = viewport.draw_text(x, y, &line);
-                if let Some(line_end) = line_end {
-                    viewport.draw_span(line_rect.x + line_rect.width.unwrap_isize(), y, &line_end);
-                }
+
+                let mut spans = vec![Span::raw(change_type_text)];
+                push_spans_from_line(line, &mut spans);
+
+                viewport.draw_line(x, y, &Line::from(spans).style(changed_line_style));
```

Comment:
note: this could be done with iterators if [`ratatui` was updated to `0.26.3` for `Line::from_iter`](https://docs.rs/ratatui/0.26.3/ratatui/text/struct.Line.html#method.from_iter)

But it doesn't look that much better, and it's more code than `push_spans_from_line`, but it could save you a `Vec` if that was desired for some reason <shrug>.